forest/daemon/src/daemon.rs

```diff
@@ -388,19 +389,41 @@ pub(super) async fn start(
         unblock_parent_process()?;
     }
 
-    // Fetch and ensure verification keys are downloaded
+    // Sets proof parameter file download path early, the files will be checked and
+    // downloaded later right after snapshot import step
     if cns::FETCH_PARAMS {
-        use forest_paramfetch::{
-            get_params_default, set_proofs_parameter_cache_dir_env, SectorSizeOpt,
-        };
-        set_proofs_parameter_cache_dir_env(&config.client.data_dir);
-
-        get_params_default(&config.client.data_dir, SectorSizeOpt::Keys).await?;
+        forest_utils::proofs_api::paramfetch::set_proofs_parameter_cache_dir_env(
+            &config.client.data_dir,
+        );
     }
 
     let config = maybe_fetch_snapshot(should_fetch_snapshot, config).await?;
 
-    sync_from_snapshot(&config, &state_manager).await?;
+    if let Some(path) = &config.client.snapshot_path {
+        let stopwatch = time::Instant::now();
+        import_chain::<_>(
+            &state_manager,
+            &path.display().to_string(),
+            config.client.skip_load,
+        )
+        .await
+        .context("Failed miserably while importing chain from snapshot")?;
+        info!("Imported snapshot in: {}s", stopwatch.elapsed().as_secs());
+    }
+
+    let validate_height = if config.client.snapshot {
+        config.client.snapshot_height
+    } else {
+        None
+    };
+    let proof_params_required = !opts.halt_after_import || validate_height.is_some();
+    if proof_params_required {
+        forest_utils::proofs_api::paramfetch::ensure_params_downloaded().await?;
+    }
```

Comment:
Instead of this logic, you can just call `ensure_params_downloaded` twice. Then the logic is encoded in the control flow.